src/vm/class.cpp

```diff
@@ -4024,43 +3770,528 @@ void EEClassLayoutInfo::ParseFieldNativeTypes(
                 szFieldName = "Invalid FieldDef record";
             }
 #endif
-
-
             MetaSig fsig(pCOMSignature, cbCOMSignature, pModule, pTypeContext, MetaSig::sigField);
             CorElementType corElemType = fsig.NextArgNormalized();
-            if (!(*fDisqualifyFromManagedSequential))
-            {
-                *fDisqualifyFromManagedSequential = IsDisqualifiedFromManagedSequential(corElemType, fsig, &pFieldInfoArrayOut->m_managedPlacement);
-            }
-
-            ParseNativeType(pModule,
-                fsig.GetArgProps(),
-                fd,
-                nativeTypeFlags,
-                &pFieldInfoArrayOut->m_nfd,
-                pTypeContext
-#ifdef _DEBUG
-                ,
-                szNamespace,
-                szName,
-                szFieldName
-#endif
-            );
+            *fDisqualifyFromManagedSequential |= CheckIfDisqualifiedFromManagedSequential(corElemType, fsig, &pFieldInfoArrayOut->m_placement);
 
-            if (!pFieldInfoArrayOut->m_nfd.IsBlittable())
-                pEEClassLayoutInfoOut->SetIsBlittable(FALSE);
+            if (!IsFieldBlittable(pModule, fd, fsig.GetArgProps(), pTypeContext, nativeTypeFlags))
+                *pIsBlittableOut = FALSE;
 
             (*cInstanceFields)++;
             pFieldInfoArrayOut++;
         }
     }
 
     _ASSERTE(i == cTotalFields);
-
-    // Set the number of fields here.
-    pEEClassLayoutInfoOut->m_numCTMFields += *cInstanceFields;
     // NULL out the last entry
     pFieldInfoArrayOut->m_MD = mdFieldDefNil;
-
 }
+
+//=======================================================================
+// Called from the clsloader to load up and summarize the field metadata
+// for layout classes.
+//
+// Warning: This function can load other classes (esp. for nested structs.)
+//=======================================================================
+#ifdef _PREFAST_
```

Comment:
You can delete these "Suppress PREFast warning about overly large function" on the functions that you are touching.

Reply:
(We do not run the analyzer to check for these anymore. And the functions are not that big anyway thanks to your refactoring.)